bookkeeper-server/src/test/java/org/apache/bookkeeper/net/NetworkTopologyImplTest.java

```diff
@@ -65,6 +65,35 @@ public void getLeavesShouldReturnNodesInScope() {
       assertTrue(leavesScopeRack1.contains(bookieRack1ScopeNode));
   }
 
+  @Test
+  public void testRestartBKWithNewRackDepth() {
+      NetworkTopologyImpl networkTopology = new NetworkTopologyImpl();
+      String dp1Rack = "/rack-1";
+      String dp2Rack = "/dp/rack-1";
+      BookieId bkId1 = BookieId.parse("bookieIdScopeRack0");
+      BookieId bkId2 = BookieId.parse("bookieIdScopeRack1");
+
+      // Register 2 BKs with depth 1 rack.
+      BookieNode dp1BkNode1 = new BookieNode(bkId1, dp1Rack);
+      BookieNode dp1BkNode2 = new BookieNode(bkId2, dp1Rack);
+      networkTopology.add(dp1BkNode1);
```

Comment:
Adding a case for the middle state will be better.
The only one bks shutdown, then restart it with depth 2 rack, it also can't add successfully.

Reply:
Added